lib/testing/run_tests.sh

```diff
@@ -7,8 +7,8 @@ TEST_RUNNER="$TEST_PROG_DIR/test_runner.sh"
 RET=0
 
 echo "Running all tests from $TESTS_DIR"
-for f in "$TESTS_DIR"/*.sh; do
-    "$TEST_RUNNER" "$f" || RET=1
+for f in "$TESTS_DIR"/*; do
+    "$TEST_RUNNER" "$f/test-$(basename $f).sh" || RET=1
```

Comment:
Took me a few tries to understand what's going on here; please split this up so it becomes more clean that $(basename $f) is the tool name.

Or maybe this could just be a two-level glob (for f in "$TESTS_DIR"/*/*.sh) ? Did you try that?

Reply:
I wanted to use files called `test-$(TOOL_NAME).sh` as an entry point to the tests (since there might be also other shell scripts), but maybe just files with a `test-` prefix?

Reply:
`for f in "$TESTS_DIR"/*/test-*.sh` then? I think it would be fine to support multiple entry points in case we want to split things up in the future...

Also, probably test that the file exists before executing it (in case the glob doesn't match anything)